src/coreclr/vm/interoplibinterface_shared.cpp

```diff
@@ -31,6 +31,9 @@ void Interop::OnGCStarted(_In_ int nCondemnedGeneration)
 #ifdef FEATURE_COMWRAPPERS
```

Comment:
Maybe should compare with `GetMaxGeneration`. 
It will be `2` for the foreseeable future, but it might be cleaner to not assume that,

Reply:
Hmmmm. I may want to chat offline about this. I think I need to get more educated in this area first.

Reply:
From what I see in the code, you check for the top generation just to be sure it is not nested. That seems correct and practically it will be 2. 
On the other hand `GetMaxGeneration` would be cheap enough to call once per-GC and not have a literal in the code.

Reply:
perhaps even simpler would be to move `Interop::OnGCStarted` invocation under `if (condemned == max_gen)` in the `GcStartWork`